builder/test/cse.cc

```diff
@@ -0,0 +1,182 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cassert>
+#include <map>
+#include <string>
+
+#include "builder/cse.h"
+#include "builder/node_mutator.h"
+#include "builder/simplify.h"
+#include "builder/substitute.h"
+#include "runtime/expr.h"
+#include "runtime/print.h"
+
+namespace slinky {
+
+#ifndef NDEBUG
+const node_context* set_default_print_context(const node_context* ctx);
```

Comment:
I'd rather not grow new dependencies on this, it's quite a hack. Just pass the context when printing?